cpp/arcticdb/util/name_validation.cpp

```diff
@@ -0,0 +1,126 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#include <arcticdb/util/name_validation.hpp>
+
+#include <arcticdb/entity/key.hpp>
+#include <arcticdb/entity/types.hpp>
+#include <arcticdb/util/configs_map.hpp>
+#include <arcticdb/stream/index.hpp>
+
+namespace arcticdb {
+
+// '*', '<' and '>' are problematic for S3
+const auto UNSUPPORTED_S3_CHARS = std::set<char>{'*', '<', '>'};
+
+// We currently require the size to fit in uint8. See entity/serialized_key.hpp
+const auto MAX_SIZE = 255;
+
+void verify_name(
+        const std::string &name_type_for_error,
```

Comment:
can we stick to,
```
const std::string& name_type_for_error
```
style please? Maybe we should add something to the dev guidelines wiki about this